### PR TITLE
Update code sample for vector layer exploration

### PR DESCRIPTION
**Pull request by DelazJ**

### CODE REVIEW

**Comment by DelazJ** [comment]
Failed to update the processing code to iterate over features. Any help is more than welcome so that the warning message moves below.

File: source/docs/pyqgis_developer_cookbook/vector.rst [hunk 1]
```diff
@@ -48,21 +46,38 @@ a :class:`QgsVectorLayer` object.
...
+     elif geomType == 'Polygon':
+         if QgsWkbTypes.isSingleType(geom.wkbType()):
+             x = geom.asPolygon()
+             print("Polygon: ", x, "Area: ", geom.area())
+         else:
+             x = geom.asMultiPolygon()
+             print("MultiPolygon: ", x, "Area: ", geom.area())
      else:
-         print("Unknown")
+         print("Unknown or invalid geometry")
```
[the remark this answers is not in the document]
Comment (Gustry) [reply source/docs/pyqgis_developer_cookbook/vector.rst]: In the legacy code, when not just changing from
```
geom.wkbType() == QgsWkbTypes.Point:
```
to
 ```
geom.type() == QgsWkbTypes.PointGeometry
```
Etc for others.

> Issue is that geom.wkbType() returns an integer

It returns an element from an enum. True if you print them, you get an integer. But you can compare to the enum it self. Like in the code snippet above.

[the remark this answers is not in the document]
Comment (DelazJ) [reply source/docs/pyqgis_developer_cookbook/vector.rst]: Thanks for this lesson, Étienne. I was sure there'd be a more obvious, simpler way to check such information but didn't know enum behavior.

File: source/docs/pyqgis_developer_cookbook/vector.rst [hunk 2]
```diff
@@ -48,21 +46,38 @@ a :class:`QgsVectorLayer` object.
...
+             print("MultiLine: ", x, "length: ", geom.length())
+
+     elif geom.type() == QgsWkbTypes.PolygonGeometry:
+         if geomSingleType:
+             x = geom.asPolygon()
+             print("Polygon: ", x, "Area: ", geom.area())
+         else:
+             x = geom.asMultiPolygon()
+             print("MultiPolygon: ", x, "Area: ", geom.area())
      else:
```
Comment (DelazJ) [inline source/docs/pyqgis_developer_cookbook/vector.rst]: How would I check a geometryless feature? i mean, using a table (a nogeometry layer) or a feature I removed its geometry in a line layer, geom.type() returns UnknownGeometry for both hence triggers this condition. Inserting a ` elif geom.type() == QgsWkbTypes.NoGeometry` condition does not seem to catch anything.
Btw, the way the attributes are exposed in https://qgis.org/pyqgis/master/core/Wkb/QgsWkbTypes.html does not really help beginners to sort out their relation.

**Pull request merged**